includes/rest-api/class-sensei-rest-api-questions-controller.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Sensei REST API: Sensei_REST_API_Questions_Controller class.
+ *
+ * @package sensei-lms
+ * @since 3.9.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+} // Exit if accessed directly.
+
+/**
+ * A REST controller for Sensei LMS question CPT.
+ *
+ * @since 3.9.0
+ *
+ * @see WP_REST_Posts_Controller
+ */
+class Sensei_REST_API_Questions_Controller extends WP_REST_Posts_Controller {
+	/**
+	 * Constructor.
+	 *
+	 * @param string $post_type Post type.
+	 */
+	public function __construct( $post_type ) {
+		parent::__construct( $post_type );
+
+		// This filter is needed in order for teachers to only see their own questions.
+		add_filter( 'rest_question_query', [ $this, 'exclude_others_questions' ], 10, 2 );
```

Comment:
I had the impression that questions are shared between users. I'll probably need to check the quiz endpoint that ownership is respected.

Reply:
I thought so, too, but it looks like we filter them out. 

One thing to note for your quiz endpoint is I don't think we change the post author on questions when a course teacher is changed. It seems flimsy... like if a question is added to a quiz on a lesson in a course that had a teacher change, that new teacher can remove the question from the quiz (or change it 😢 ), but they can't re-add it.